adapters/ix/ix.go

```diff
@@ -417,6 +420,28 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
 				}
 			}
 
+			var bidNative1_1 *Native11Wrapper
+			if bidType == openrtb_ext.BidTypeNative {
+				unmarshalExtErr := json.Unmarshal([]byte(bid.AdM), &bidNative1_1)
+				if unmarshalExtErr == nil && len(bidNative1_1.Native.EventTrackers) > 0 {
+					mergeNativeImpTrackers(&bidNative1_1.Native)
+					if json, marshalErr := json.Marshal(bidNative1_1); marshalErr == nil {
+						bid.AdM = string(json)
+					}
```

Comment:
@manigandham thank you for adding all of those test cases. In the case of the very improbable `json.Marshal` error, could we add that error to the `errs` array instead? Only if you think it makes sense, of course.
```
361 func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
362     switch {
363     case response.StatusCode == http.StatusNoContent:
364         return nil, nil
              .
              .
              .
399
400     var errs []error
401
402     for _, seatBid := range bidResponse.SeatBid {
              .
              .
              .
423             var bidNative1_1 *Native11Wrapper
424             if bidType == openrtb_ext.BidTypeNative {
425                 unmarshalExtErr := json.Unmarshal([]byte(bid.AdM), &bidNative1_1)
426                 if unmarshalExtErr == nil && len(bidNative1_1.Native.EventTrackers) > 0 {
427                     mergeNativeImpTrackers(&bidNative1_1.Native)
428 -                   if json, marshalErr := json.Marshal(bidNative1_1); marshalErr == nil {
429 -                       bid.AdM = string(json)
430 -                   }
    +                   json, marshalErr := json.Marshal(request)
    +                   if marshalErr != nil {
    +                   	errs = append(errs, marshalErr)
    +                   	continue //<-- if you think this makes sense in your adapter's context, of course
    +                   }
431                 }
432             }
433
434             var bidNative1_2 *native1response.Response
435             if bidType == openrtb_ext.BidTypeNative {
436                 unmarshalExtErr := json.Unmarshal([]byte(bid.AdM), &bidNative1_2)
437                 if unmarshalExtErr == nil && len(bidNative1_2.EventTrackers) > 0 {
438                     mergeNativeImpTrackers(bidNative1_2)
439 -                   if json, marshalErr := json.Marshal(bidNative1_2); marshalErr == nil {
440 -                       bid.AdM = string(json)
441 -                   }
    +                   json, marshalErr := json.Marshal(request)
    +                   if marshalErr != nil {
    +                   	errs = append(errs, marshalErr)
    +                   	continue //<-- if you think this makes sense in your adapter's context, of course
    +                   }
442                 }
443             }
              .
              .
              .
451     }
452
453     return bidderResponse, errs
454 }
adapters/ix/ix.go
```

Reply:
Do you mean adding both the unmarshall and marshal errors?

Reply:
I'd like to skip this if that's alright with you? We intend this to be short-term as we're fixing the responses coming from our our exchange soon. At that point we'll delete this workaround code anyway. Let me know...

Reply:
@manigandham we discussed and we're ok with skipping this.

Reply:
@manigandham we discussed and we're ok with skipping this.

Reply:
Thanks! I believe that's all the fixes you asked for unless there's anything else...